File: MartaCard.js
var card=args.queryParameters.card;
var name=args.queryParameters.name;
var js="";
var sentence="";

if (name==undefined){
  name="Russell";
}
if (card==undefined){
  card="Dummy"; //default card value, don't commit with value
}

var url="https://balance.breezecard.com/breezeWeb/cardnumber_qa.do?submitButton.x=0&submitButton.y=0&cardnumber=";
url=url+card;
let wv=new WebView();
await wv.loadURL(url, true);

// build function
js='function get(){ return document.querySelectorAll("b")[0].innerHTML;} get();';
let balanceProtect=await wv.evaluateJavaScript(js, false);

js='function get(){ return document.querySelectorAll("b")[2].innerHTML;} get();';
let hotListed=await wv.evaluateJavaScript(js, false);

js='function get(){ return document.querySelectorAll("td")[8].innerHTML;} get();';
let expiration=await wv.evaluateJavaScript(js, false);
expiration=expiration.replace(/-/g,"/").replace("  : ","");

js='function get(){ return document.querySelectorAll("td")[document.querySelectorAll("td").length-9].innerHTML;} get();';
let cash=await wv.evaluateJavaScript(js, false);
cash=cash.replace("$","");
cash=cash.replace("0.00","zero");

js='function get(){ return document.getElementsByClassName("Content_normal_black")[document.getElementsByClassName("Content_normal_black").length-1].innerHTML;} get();';
let pending=await wv.evaluateJavaScript(js, false);

js='function get(){ var trips="zero"; if(document.querySelector("[align=center]")!=null){ trips=0; for(var i=0; i<document.querySelectorAll("[align=center]").length-1; i++){ trips=trips+parseInt(document.querySelectorAll("[align=center]")[i].innerHTML); } } return trips; } get();';
let trips=await wv.evaluateJavaScript(js, false);

log("balanceProtect:"+balanceProtect);
log("hotListed:"+hotListed);
log("cash:"+cash);
log("expiration:"+expiration);
log("pending:"+pending);
log("trips:"+trips);

sentence=sentence+"Hi "+name+", you have "+trips+" trips remaining, with a cash value of "+cash+" dollars. "+expiration;

if (balanceProtect.indexOf("Yes")>-1){
  sentence=sentence+" and is balance protected.";
}
if (balanceProtect.indexOf("No")>-1){
  sentence=sentence+" and is Not balance protected.";
}

if (hotListed.indexOf("Yes")>-1){
  sentence=sentence+" Your card is hot listed.";
}
if (hotListed.indexOf("No")>-1){
  sentence=sentence+" Your card is not hot listed.";
}

log(sentence);
Safari.open("shortcuts://x-callback-url/run-shortcut?name=SpeakScript&input="+encodeURI(sentence));

File: MartaCard.py
// This is code for the pythonista iOS app.
// You can call it from a shortcut by opening URL 
// pythonista:///MartaCard.py&argv=12341234123412341234&argv=Russell
// note the arguments



import urllib, bs4, requests, sys, webbrowser, console
console.clear()
def get_beautiful_soup(url):
    return bs4.BeautifulSoup(requests.get(url).text,'html5lib')
soup=get_beautiful_soup('https://balance.breezecard.com/breezeWeb/cardnumber_qa.do?submitButton.x=0&submitButton.y=0&cardnumber='+str(sys.argv[1]))
trips=str(soup.find('div',attrs={"align":"center"}).text).strip(" ")
s="Hi "+str(sys.argv[2])+", you have "+trips+" trips remaining, "
if not trips:
	s="Hi "+str(sys.argv[2])+", you have zero trips remaining, "
s=s+" with a cash value of "+str(soup.find_all('td')[18].text)
s=s+".  "+str(soup.find_all('td',attrs={"class":"Content_bold"})[1].text).replace('-','/').replace('  : ','')
if str(soup.find('b').text)=='Yes ':
  s=s+' and is Balance Protected.'
else:
	s=s+' and is NOT Balance Protected.'
if str(soup.findAll('b')[2].text)=='No':
  s=s+'  Your card is not hot listed.'
else:
	s=s+'  Your card is hot listed.'
s=s+" You have "+str(soup.find_all('td',attrs={"class":"Content_normal_black"})[3].text)+".  Would you like to view your account on the web?"+'_'+str(sys.argv[1])
webbrowser.open('shortcuts://x-callback-url/run-shortcut?name=SpeakMarta&input='+urllib.parse.quote(s))






